apm-sniffer/apm-sdk-plugin/vertx-plugins/vertx-core-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/vertx4/SWVertxTracer.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.vertx4;
+
+import io.vertx.core.Context;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.eventbus.impl.clustered.ClusteredMessage;
+import io.vertx.core.impl.ContextInternal;
+import io.vertx.core.spi.observability.HttpRequest;
+import io.vertx.core.spi.observability.HttpResponse;
+import io.vertx.core.spi.tracing.SpanKind;
+import io.vertx.core.spi.tracing.TagExtractor;
+import io.vertx.core.spi.tracing.VertxTracer;
+import io.vertx.core.tracing.TracingPolicy;
+import org.apache.skywalking.apm.agent.core.context.CarrierItem;
+import org.apache.skywalking.apm.agent.core.context.ContextCarrier;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.ContextSnapshot;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+public class SWVertxTracer implements VertxTracer<AbstractSpan, AbstractSpan> {
+
+    @Override
+    public <R> AbstractSpan receiveRequest(Context context, SpanKind kind, TracingPolicy policy, R request,
+                                           String operation, Iterable<Map.Entry<String, String>> headers,
+                                           TagExtractor<R> tagExtractor) {
+        if (TracingPolicy.IGNORE.equals(policy)) {
+            return null;
+        }
+
+        if (request instanceof HttpRequest) {
+            HttpRequest serverRequest = (HttpRequest) request;
+
+            ContextCarrier contextCarrier = getContextCarrier(headers);
+            AbstractSpan span = toEntrySpan(
+                    "{" + serverRequest.method() + "}" + serverRequest.uri(),
```

Comment:
According to latest operation name style, this should be `method` + `:` + `URI`, referring from 

https://github.com/apache/skywalking-java/blob/2bd58bce7a95e61eac810ecb6340beaa3dfe3892/apm-sniffer/apm-sdk-plugin/spring-plugins/mvc-annotation-commons/src/main/java/org/apache/skywalking/apm/plugin/spring/mvc/commons/interceptor/AbstractMethodInterceptor.java#L263

This is following OpenAPI style, https://skywalking.apache.org/docs/main/v8.9.1/en/setup/backend/endpoint-grouping-rules/#endpoint-name-grouping-by-openapi-definitions